controllers/virtinkmachine_controller.go

```diff
@@ -56,42 +59,90 @@ func (r *VirTinkMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, fmt.Errorf("create Machine patch helper: %s", err)
 	}
 
-	if err := r.reconcile(ctx, &machine); err != nil {
+	res, err := r.reconcile(ctx, &machine)
+	if err != nil {
 		return ctrl.Result{}, err
 	}
 
 	if err := patchHelper.Patch(ctx, &machine); err != nil {
 		return ctrl.Result{}, fmt.Errorf("patch Machine: %s", err)
 	}
-	return ctrl.Result{}, nil
+	return res, nil
 }
 
-func (r *VirTinkMachineReconciler) reconcile(ctx context.Context, machine *infrastructurev1beta1.VirTinkMachine) error {
-	if !machine.DeletionTimestamp.IsZero() {
-		return nil
+func (r *VirTinkMachineReconciler) reconcile(ctx context.Context, machine *infrastructurev1beta1.VirTinkMachine) (ctrl.Result, error) {
```

Comment:
I prefer build the client outside this func and then call this func with the client ref. Or better, build the client here and call reconcileWithClient with the client.